### ISSUE TITLE
Type error when using reduce()

### ISSUE THREAD

**Issue by ljhhuxiaoba** [neo4j/neo4j#13189]
I execute a query on my neo4j database ,and an error occurs.

Neo4j version: 5.7.0 Community
operating system: windows 10
API：Cypher version 9
query:match (a) return [a][count{(a)}] limit reduce(a=1,b in [1]|1)
error message:
Neo.ClientError.Statement.TypeError
Expected a Node, got: Long(1)

But if I execute another query such as "match (a) return [a][count{(a)}] limit reduce(c=1,b in [1]|1)",then it can run successfully and no error occurs.

Expected behavior:
The query should run successfully.

Actual behavior:
The query failed.

**Comment by arnefischereit** [comment]
Hi @ljhhuxiaoba 

I just tried to run the query that you wrote `match (a) return [a][count{(a)}] limit reduce(a=1,b in [1]|1)` on an empty docker image neo4j:5.7 and that one succeeded.

Could you please verify that this particular query fails on 5.7.0 for you? Can you try to reproduce in a docker image?

Regards,

Arne

**Comment by ljhhuxiaoba** [comment]
I just ran the query on my empty neo4j database:5.7 community(not docker) and this bug seems to have just been fixed, as it did exist when the issue was submitted a few days ago.

**Comment by arnefischereit** [comment]
So, what version did you try it with before?

**Comment by ljhhuxiaoba** [comment]
I am so sorry for forgetting to reply to your question and the version I try with is 5.7 community(not docker) just as mentioned in the issue.

**Comment by arnefischereit** [comment]
Hi @ljhhuxiaoba,

Any news on this issue?

**Comment by ljhhuxiaoba** [comment]
The same TypeError occurs on 5.8 community again.

**Comment by arnefischereit** [comment]
Could you please run

```
EXPLAIN
match (a) return [a][count{(a)}] limit reduce(a=1,b in [1]|1)
```

for me?

Also, it would be interesting what your database looks like. Is it empty or do you have data in there?

What is the output of

```
CALL db.stats.retrieve('GRAPH COUNTS')
```

?

Do you get any stack-traces in the `debug.log`?

**Comment by ljhhuxiaoba** [comment]
This is the query plan:
![屏幕截图 2023-06-13 223746](https://github.com/neo4j/neo4j/assets/67541048/f1915ec3-77d8-4dc4-8b8b-40cf829342cc)

And my database has only one node(by running "create ()").But when I delete the node,then the error disappears!very interesting.

**Comment by ljhhuxiaoba** [comment]
![屏幕截图 2023-06-13 225004](https://github.com/neo4j/neo4j/assets/67541048/e9f26992-b2dc-4b29-83c2-0447f0ec562a)

This is the output of "CALL db.stats.retrieve('GRAPH COUNTS')",I hope the above information can help you.

**Comment by arnefischereit** [comment]
Yes, they do indeed: I usually assume an empty database when trying out bugs, so I did not have that one node. Now I can reproduce the issue.

**Comment by ljhhuxiaoba** [comment]
This is also my problem due to forgetting to provide database data.

**Comment by arnefischereit** [comment]
Thanks for reporting the bug and for your persistence in helping me reproduce it. We will investigate and let you know the outcome.

---

[Link for internal reference](https://trello.com/c/yLNHuuSY/)

**Comment by Lojjs** [comment]
@ljhhuxiaoba The bug was related to a scoping issue with SKIP and LIMIT and has now been fixed (same issue as on 2 of your other Github issues). The fix will be included in upcoming Neo4j 5.11 and 4.4.24 releases. 

Best regards, Louise, Neo4j Cypher team
